app/scripts/views/mixins/open-confirmation-email-mixin.js

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+/**
+ * A view mixin that helps confirm* views show the "Open Gmail" button
+ * directly from FxA.
+ */
+define(function (require, exports, module) {
+  'use strict';
+
+  return {
+    events: {
+      'click #open-gmail': '_gmailTabOpened',
+      'click #open-hotmail': '_hotmailTabOpened',
+      'click #open-yahoo': '_yahooTabOpened'
+    },
+
+    /**
+     * Check if the `Open Gmail` button should be visible
+     *
+     * @param {string} email
+     * @returns {boolean}
+     */
+    isOpenGmailButtonVisible: function (email) {
+      // The "Open Gmail" is only visible in certain contexts
+      // we do not show it in mobile contexts because it performs worse on mobile
+      return this.broker.hasCapability('openGmailButtonVisible') &&
+             /@gmail\.com$/.test(email);
+    },
+
+    isOpenHotmailButtonVisible: function (email) {
+      // The "Open Hotmail" is only visible in certain contexts
+      // we do not show it in mobile contexts because it performs worse on mobile
+      return this.broker.hasCapability('openHotmailButtonVisible') &&
+             /@hotmail\.com$/.test(email);
+    },
+
+    isOpenYahooButtonVisible: function (email) {
+      // The "Open Yahoo" is only visible in certain contexts
+      // we do not show it in mobile contexts because it performs worse on mobile
+      return this.broker.hasCapability('openYahooButtonVisible') &&
+             /@yahoo\.com$/.test(email);
```

Comment:
instead of 3-6 functions, this can just use an object structure or an array that has the `url`, `buttonText` and maybe some `id`/`class` value.